tests/plugin/test_playlist.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2022 Felix Krull
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+import os
+from textwrap import dedent
+
+from quodlibet.formats import AudioFile
+
+from tests.plugin import PluginTestCase
+from tests.helper import temp_filename
+
+
+class TPlaylistExport(PluginTestCase):
+    def setUp(self):
+        self.mod = self.modules["Playlist Export"]
+
+    def test_m3u_playlist(self):
+        plugin = self.mod.PlaylistExport()
+        song1 = AudioFile(
+            {
+                "~filename": "/a/b/c.mp3",
+                "~#length": 123,
+                "artist": "a",
+                "title": "c",
+            }
+        )
+        song2 = AudioFile(
+            {
+                "~filename": "/a/b/d.mp3",
+                "~#length": 400,
+                "artist": "b",
+                "title": "d",
+            }
+        )
+
+        with temp_filename() as playlist_file_path:
+            plugin.save_playlist(
+                [song1, song2], playlist_file_path, self.mod.FORMAT_M3U, relative=False
+            )
+            with open(playlist_file_path, "r") as f:
+                result = f.read()
+
+        self.assertEqual(
+            result,
+            dedent(
+                """\
+            #EXTM3U
+            #EXTINF:123,a - c
+            /a/b/c.mp3
+            #EXTINF:400,b - d
+            /a/b/d.mp3
+            """
+            ),
+        )
+
+    def test_m3u_playlist_relative(self):
+        plugin = self.mod.PlaylistExport()
+        with temp_filename() as playlist_file_path:
+            d = os.path.dirname(playlist_file_path)
+            pd = os.path.dirname(d)
+            song1 = audio_file(
+                filename=os.path.join(d, "a", "b.mp3"), length=23, artist="a", title="b"
+            )
+            song2 = audio_file(
+                filename=os.path.join(pd, "c.mp3"),
+                length=1,
+                artist="a",
+                title="c",
+            )
+
+            plugin.save_playlist(
+                [song1, song2], playlist_file_path, self.mod.FORMAT_M3U, relative=True
+            )
+            with open(playlist_file_path, "r") as f:
```

Comment:
And this is just `playlist_file_path.read_text()` :up:

Reply:
:heavy_check_mark: